### PR TITLE
Use stored proc for specific queries

### PR DESCRIPTION
**Pull request by feordin**
## Description
This creates a store of hash values representing known problematic queries and introduces a stored procedure for executing them.

## Related issues
Addresses [issue #].

## Testing
Describe how this change was tested.

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.SqlServer/Features/Search/CustomQueries.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+using System.Collections.Generic;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Search
+{
+    public static class CustomQueries
+    {
+        public static readonly Dictionary<string, string> QueryStore = new Dictionary<string, string>()
+            {
+                { "7100AADC5527EC3D808DB1A0F9F8C3B71522683D0F8EE1F0C1DA3E4570FE586B", "dbo.QueryIdentifierWithIncludeAndOtherParam"},
```
Comment (SergeyGaluzo) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Search/CustomQueries.cs:14]: >QueryIdentifierWithIncludeAndOtherParam [](http://example.com/codeflow?start=91&length=39)

I would suggest to use hash in the name of store procedure. Like ExceptionSearch_<hash>. And have this list updated from the database.

**Comment by SergeyGaluzo** [comment]
Let's make this logic generic.

**Review by SergeyGaluzo — CHANGES_REQUESTED** [review]
:clock1:

**Comment by feordin** [comment]
Have merged these changes with PR https://github.com/microsoft/fhir-server/pull/3207.  Closing this PR without merging.

**Pull request closed without merging**